lib/puppet/provider/cisco_vxlan_global/nxapi.rb

```diff
@@ -46,7 +46,7 @@
 
   def initialize(value={})
     super(value)
-    @vxlan_global = Cisco::VxlanGlobal.new
+    @vxlan_global = Cisco::VxlanGlobal.new if Cisco::VxlanGlobal.enabled
```

Comment:
ensure: absent Idempotence beaker test for default properties was always having to remove the resource. Thats why the test was failing with exit code 2, whereas the idempotence test expects always 0.

The reason was in provider initialize, a new instance was always getting created even the resource was disabled and therefore the ensure:absent always removed it which was incorrect. This check fixes it.